pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiQuestionValidation.java

```diff
@@ -61,7 +61,7 @@
         Map<Long, List<RuleDto>> questionIdToRuleDtos = new HashMap<>();
         try (var stream = getAllActiveValidationDtosForActivity(activityId)) {
             stream.forEach(ruleDto -> questionIdToRuleDtos
-                    .computeIfAbsent(ruleDto.getQuestionId(), k -> new ArrayList<>())
+                    .computeIfAbsent(activityId, k -> new ArrayList<>())
```

Comment:
I think the current code is correct. I think this `getAllActiveValidationsForActivity()` method is meant to find all `RuleDto`s for the given activity id, but grouped by question id. So the resulting Map is `(question_id -> list)`.

This Map is then added to the `questionIdToValidationsCache`, which will accumulate all these `RuleDto`s for all the questions since questions across activities have unique question ids.

Reply:
The problem was that for different questions the result was not found in the cache (in the cache the key should be activity ID). But it was searched in cache by question ID. 
Seems you are right, the code was correct but not quite correct. I'll refix it.

Reply:
@yufengwng I remembered what was a problem: it is not actually correct to store in Redis cache the question validations by a question ID. Because there are questions without validation. And if we query a cache by such question ID then it's validation are not found (which is quite correct - it not defined for this question) but the next step: the SQL query 'getAllActiveValidationsForActivity' is executed again.
The correct way - to store a list of question validations for all activity with key = activity ID.
If you ok with it then I will try to fix the implementation (rename 'questionIdToValidationsCache' etc)..

Reply:
Okay, I see. So I'll say in this case the code is correct -- it was doing what it was designed for. But this design is not helping us because it results in cache misses for scenarios where question doesn't actually have validations. Going up a level and caching by activity id sounds reasonable to me. Go for it!

Reply:
@youfeng, I reimplemented the more proper way. Now the only class changed is JdbiQuestionValidationCached.java:
key = activityId, value = Map<Long, List<RuleDto>> (where key = questionId).
This solved the problem with repeated SQL query.